helios-core/src/main/kotlin/helios/core/errors.kt

```diff
@@ -1,10 +1,20 @@
 package helios.core
 
+import arrow.core.Option
+import java.time.format.DateTimeFormatter
+
 sealed class DecodingError
-data class StringDecodingError(val value: Json) : DecodingError()
-data class BooleanDecodingError(val value: Json) : DecodingError()
-data class NumberDecodingError(val value: Json) : DecodingError()
-data class ArrayDecodingError(val value: Json) : DecodingError()
-data class ObjectDecodingError(val value: Json) : DecodingError()
+data class JsNullDecodingError(val value: Json) : DecodingError()
+data class JsStringDecodingError(val value: Json) : DecodingError()
+data class JsBooleanDecodingError(val value: Json) : DecodingError()
+data class JsNumberDecodingError(val value: Json) : DecodingError()
```

Comment:
Should we consider making this a nested ADT mimicking the `Json` structure? That would offer more fine grained control over number errors which can be tedious to debug when needed. Fine to do in another issue.

Reply:
Done